src/utils/hash.hpp

```diff
@@ -19,51 +19,75 @@
 #define UTILS_HASH_HPP_
 
 #include <functional>
+#include <set>
 #include <tuple>
 #include <utility>
+#include <vector>
 
 namespace parthenon {
+
+// A hash struct that can be used as a template class in
```

Comment:
I trust that the updates here work as expected (probably refactored 1:1 from boost, aren't they?)